### PR TITLE
Created new data sample

### LINKED ISSUE #1546: Update the Sample Data With More Meaningful Entries
We need to have more realistic data in the Talawa-API sample data importation process. The data also needs to be more secular so that the application will more likely appeal to a wider range of Admins testing Talawa for their communities.

The organization names also need to reflect the most common use case of related organizations being managed by Talawa. They also need to be more global to have broader appeal

Therefore the following changes are needed.

## Possible Approach

You will need to investigate whether this approach is viable and suggest modifications where applicable.

1. Load the current sample data into Talawa Admin
2. Use Talawa Admin to make the modifications below
3. Save the resulting database as the new sample data that `setup.ts` uses

## New Organization Names

Rename the organizations in the sample data to the following, with updated geographic location data. These cities are well known and geographically close together. This would be a typical scenario for a community based organization

1. Unity Foundation – Bronx
1. Unity Foundation – Queens
1. Unity Foundation – Staten Island
1. Unity Foundation – Brooklyn

## Users

We will need more users who can be used in later tasks

1. Add 8 new users
1. They must be registered members for all 4 organizations

## Events

We need recurring events created so that the calendar is populated no matter when the data is loaded.

### Weekly Recurring Events – Non Registration

Add the following public events to each organization. Registration is not required:

1. Monday:
    1. "Aerobics for Everyone": 6:00am – 7:00am, Location: Studio A
    1. "Early Morning Pilates": 6:00am – 7:00am, Location: Studio B
    1. "Yoga Class for Seniors": 10:00am – 11:00am, Location: Studio B
    1. "Lap Swimming": 6:00am – 3:00pm, Location: Pool
    1. "Book of the Week Club": 2:00pm – 3:00pm, Location: Atlantic Conference Room
1. Tuesday:
    1. "Aerobics for Everyone": 6:00am – 7:00am, Location: Studio A
    1. "Early Morning Pilates": 6:00am – 7:00am, Location: Studio B
    1. "Travel Club": 6:00pm – 7:00pm, Location: Atlantic Conference Room
    1. "Lap Swimming": 6:00am – 3:00pm, Location: Pool
1. Wednesday:
    1. "Aerobics for Everyone": 6:00am – 7:00am, Location: Studio A
    1. "Early Morning Pilates": 6:00am – 7:00am, Location: Studio B
    1. "Travel Club": 6:00pm – 7:00pm, Location: Atlantic Conference Room
    1. "Art Class - Advanced": 10:00am – 11:00am, Location:
    1. "Lap Swimming": 6:00am – 3:00pm, Location: Pool
1. Thursday:
    1. "Aerobics for Everyone": 6:00am – 7:00am, Location: Studio A
    1. "Early Morning Pilates": 6:00am – 7:00am, Location: Studio B
    1. "Kids Karate": 6:00pm – 7:00pm, Location: Studio B
    1. "Art Class - Beginners": 3:00pm – 4:00pm, Location: Studio A
    1. "Lap Swimming": 6:00am – 3:00pm, Location: Pool
1. Friday:
    1. "Aerobics for Everyone": 6:00am – 7:00am, Location: Studio A
    1. "Early Morning Pilates": 6:00am – 7:00am, Location: Studio B
    1. "Neighborhood Watch Meeting": 6:00pm – 7:00pm, Location: Atlantic Conference Room
    1. "Lap Swimming": 6:00am – 3:00pm, Location: Pool
1. Saturday:
    1. "So Bad They’re Good Movie Night": 7:00pm – 9:00pm, Location: Main Theatre
1. Sunday:
    1. "Travel Club": 6:00pm – 7:00pm, Location: Atlantic Conference Room

### Weekly Recurring Events – With Registration

Add the following public events to each organization. Registration is required. Make at least 4 of the newly added users be pre-registered in each of the events with no timing conflicts.

1. Monday:
    1. "Youth Swimming Lessons": 3:00pm – 5:00pm, Location: Pool
    1. "Finance Committee Meeting": 5:00pm – 6:30pm, Location: Pacific Conference Room
    1. "Risk Committee Meeting": 6:00pm – 7:30pm, Location: Arctic Conference Room
1. Tuesday:
    1. "Adult Swimming Lessons": 3:00pm – 5:00pm, Location: Pool
    1. "Conference Planning Committee": 5:00pm – 6:30pm, Location: Pacific Conference Room
    1. "Organization Strategy Workshop": 6:00pm – 7:30pm, Location: Arctic Conference Room
1. Wednesday:
    1. "Youth Swimming Lessons": 3:00pm – 5:00pm, Location: Pool
    1. "Families With Addiction": 5:00pm – 6:30pm, Location: Pacific Conference Room
    1. "Investment Strategy Committee": 6:00pm – 7:30pm, Location: Arctic Conference Room
1. Thursday:
    1. "Adult Swimming Lessons": 3:00pm – 5:00pm, Location: Pool
    1. "Programming for Everyone": 3:00pm – 5:00pm, Location: Computer Lab
    1. "University Preparation Tips": 5:00pm – 8:00pm, Location: Arctic Conference Room
1. Friday:
    1. "Career Counseling": 3:00pm – 5:00pm, Location: Pacific Conference Room
1. Saturday:
    1. "Saturday Serenade Concert": 5:00pm – 7:00pm, Location: Garden Patio
    1. "Youth Soccer Lessons": 9:00am – 10:00am, Location: Sports Field
1. Sunday:
    1. "Travel Club": 6:00pm – 7:00pm, Location:

## Posts

Make all 8 new users have posts created at the time of data import for these organizations. The posts must have meaningful titles and text that matches the title (at least 20 words). Half the posts must have images, the rest must be text only. The images must be single color royalty-free and a part of the repo in an area dedicated for the sample data importation.

1. Unity Foundation – Bronx
1. Unity Foundation – Queens

## Other
1. All tests must pass
1. No functionality of the app must be affected
1. The suggested entries above may have errors. Wherever possible there must not be scheduling conflicts for:
    1. Locations
    1. Users 
 1. The sample data must be importable by the setup script without error and fully functional no matter how many times the setup script is run.

### PR DESCRIPTION
**Pull request by DecodeAndCode**
**What kind of change does this PR introduce?**
Created new data samples that are more meaningful.

**Issue Number:**

Fixes #1546

**Did you add tests for your changes?**
No

**Snapshots/Videos:**

https://github.com/PalisadoesFoundation/talawa-api/assets/107995043/919a0c4f-4959-49ed-8089-21bd4eade1cd

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**
Yes

### CODE REVIEW

File: tests/resolvers/Query/actionItemsByOrganization.spec.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ import { beforeAll, afterAll, describe, it, expect } from "vitest";
 import type mongoose from "mongoose";
 import { createTestActionItems } from "../../helpers/actionItem";
 import type { TestEventType } from "../../helpers/events";
-import { TestOrganizationType } from "../../helpers/userAndOrg";
+import type { TestOrganizationType } from "../../helpers/userAndOrg";
```
Comment (palisadoes) [inline tests/resolvers/Query/actionItemsByOrganization.spec.ts:10]: Why is this change needed?

File: src/utilities/encodedImageStorage/uploadEncodedImage.ts [hunk 2]
```diff
@@ -14,11 +14,11 @@ const checkImageSizeLimit = (size: number): boolean => {
 
 const base64SizeInKb = (base64String: string): number => {
   // Count the number of Base64 characters
-  var numBase64Chars = base64String.length;
+  const numBase64Chars = base64String.length;
```
Comment (palisadoes) [inline src/utilities/encodedImageStorage/uploadEncodedImage.ts:17]: Why are these changes needed?

File: sample_data/organizations.json [hunk 3]
```diff
@@ -2,24 +2,34 @@
...
     "address": {
-      "city": "Delhi",
-      "countryCode": "IN",
+      "city": "Bronx",
+      "countryCode": "US",
       "dependentLocality": "Some Dependent Locality",
       "line1": "123 Random Street",
       "line2": "Apartment 456",
-      "postalCode": "110001",
+      "postalCode": "110011",
```
Comment (palisadoes) [inline sample_data/organizations.json:30]: US ZIP codes are 5 digits, use valid ones for the cities, in all cases

**Review by palisadoes — CHANGES_REQUESTED** [review]
See comments

File: sample_data/organizations.json (hunk 3 above)
Comment (DecodeAndCode) [reply sample_data/organizations.json:30]: sure

File: tests/resolvers/Query/actionItemsByOrganization.spec.ts (hunk 1 above)
Comment (DecodeAndCode) [reply tests/resolvers/Query/actionItemsByOrganization.spec.ts:10]: These changes appeared on linting. Thought them as neccessary. Are they neccessary or shall I remove them?

File: src/utilities/encodedImageStorage/uploadEncodedImage.ts (hunk 2 above)
Comment (DecodeAndCode) [reply src/utilities/encodedImageStorage/uploadEncodedImage.ts:17]: Same as above

**Comment by palisadoes** [comment]
Please fix the linting. We are working on fixing the git hooks

**Review by SiddheshKukade — CHANGES_REQUESTED** [review]

File: sample_data/users.json [hunk 4]
```diff
@@ -166,5 +166,237 @@
...
+    "membershipRequests": [],
+    "organizationsBlockedBy": [],
+    "status": "ACTIVE",
+    "pluginCreationAllowed": true,
+    "adminApproved": false,
+    "firstName": "Teresa",
+    "lastName": "Bradley",
+    "email": "testuser4@example.com",
+    "password": "$2a$12$bSYpay6TRMpTOaAmYPFXku4avwmqfFBtmgg39TabxmtFEiz4plFtW",
+    "image": null,
```
[the remark this answers is not in the document]
Comment (DecodeAndCode) [reply sample_data/users.json:195]: do I have to put direct url or download and add the image to folder for both user.json and posts.json

[the remark this answers is not in the document]
Comment (palisadoes) [reply sample_data/users.json:195]: This is a good idea

1. Upload the images. URLs are unreliable
2. Place them in sub-directories under the data directory
3. Have separate sub-directories for users, posts. For example:
    1. `images/posts`
    2. `images/users`

[the remark this answers is not in the document]
Comment (DecodeAndCode) [reply sample_data/users.json:195]: Sure

[the remark this answers is not in the document]
Comment (DecodeAndCode) [reply sample_data/users.json:195]: If I am putting images address like this "images/posts/download.png" its not displaying anything on screen even url is not working both are displaying nothing on screen??

[the remark this answers is not in the document]
Comment (palisadoes) [reply sample_data/users.json:195]: If that doesn't work. Just leave it at Null. We need this PR merged

[the remark this answers is not in the document]
Comment (DecodeAndCode) [reply sample_data/users.json:195]: @palisadoes Then I guess nothing to be done from my side just reviews are required or if there is please let me know.

**Comment by palisadoes** [comment]
Please fix the failing tests

**Comment by DecodeAndCode** [comment]
@palisadoes its done check once.

**Review by SiddheshKukade — DISMISSED** [review]
LGTM. We can tackle the images part later.

**Review by palisadoes — CHANGES_REQUESTED** [review]
Setup fails. Please fix

image

**Comment by DecodeAndCode** [comment]
@palisadoes Please check now

**Comment by palisadoes** [comment]
1. Same problem. 
2. Run the setup script on your local machine to  verify
3. Check to make sure that all functions in Talawa Admin work too. We've had major disruptions because of failed data imports in the past

**Comment by palisadoes** [comment]
Please ask the talawa-api slack channel for advice

**Comment by DecodeAndCode** [comment]
@palisadoes It's working fine for me now I just changed the mongoDb url to localhost url instead of cloud instance url.

Screenshot 2024-02-15 at 12 57 31 PM

**Comment by palisadoes** [comment]
*NOTE* Read very carefully

1. We just merged this PR which upgraded the prettier package. 
    1. https://github.com/PalisadoesFoundation/talawa-api/pull/1840
1. It reformatted over 400 files. This will put your PR at risk of extensive merge conflicts.    
1. I suggest that before you merge your code to the latest upstream, or manually fix these merge conflicts that you:
    1. upgrade your `prettier` in your local branch to the same version
    1. run `prettier`
    
This will help to reduce the number of future merge conflicts for your PR.

**Comment by DecodeAndCode** [comment]
@palisadoes Did the same.
